### ISSUE TITLE
Devtools

### ISSUE THREAD

**Issue by lishine** [react-spring/zustand#104]
I try to use devtools, with the store and the reducer.
With the reducer it works but does not prepend the id.
With the store it does not work.
Also notice I try to use it as local store.
Also eventually I would like to use all store instances with the same redux instance so that to see all actions in the same list.
https://codesandbox.io/s/zustand-z06bb
Thanks

**Comment by olavoparno** [comment]
@lishine did you manage to get all instances within just one redux instance? I also am experiencing this issue.

**Comment by lishine** [comment]
Nope. I stopped trying this lib after seeing it not maintained. I usually like to use easy-peasy

**Comment by clawoflight** [comment]
@drcmda can you please, please help us with this?

**Comment by olavoparno** [comment]
> @drcmda can you please, please help us with this?
> 
> Although, @lishine: I think all you need to do is add the optional second parameter to your 'set' calls to give them a name, then it should work.

hey, dear @clawoflight, it does really work but separated by multiple Redux Devtools Instances. Lemme show you an example from my stores:

![image](https://user-images.githubusercontent.com/7513162/83338345-f1e16d00-a299-11ea-80d8-b7eabb69316f.png)

all Instance 2 to 5 belong to the same project called `Oraculo`. Each instance logs a specific store.

**Comment by clawoflight** [comment]
@olavoparno Sorry, I was on holidays.

Of course they are separated - they are completely different stores, possibly with overlapping values inside. Their is no sensible way to combine them.
However, I use a patch that shows the prefix ('platform') in your example instead of 'instance #'.

I will open a PR to upstream that once @drcmda is back.

**Comment by clawoflight** [comment]
@drcmda @JeremyRH can I maybe become co-maintainer?
It would already be very helpful if I was allowed to triage bugs - I have already replied to many of the open ones and know that quite a few could be closed.

**Comment by olavoparno** [comment]
> @olavoparno Sorry, I was on holidays.
> 
> Of course they are separated - they are completely different stores, possibly with overlapping values inside. Their is no sensible way to combine them.
> However, I use a patch that shows the prefix ('platform') in your example instead of 'instance #'.
> 
> I will open a PR to upstream that once @drcmda is back.

Thank you very much for your elightment, sir. Glad you replied and I can only hope they allow you co-maintaining this repo.

**Comment by dkgrieshammer** [comment]
For me the following hint made it work; the trick is to add names to the functions in your store
https://github.com/react-spring/zustand/issues/77#issuecomment-551435696

**Comment by sanojsilva** [comment]
This is my codebase with redux devtools but it doesn't show me the state updates

```js
import create from "zustand";
import produce from "immer";
import { devtools } from "zustand/middleware";

const immer = (config) => (set, get, api) =>
  config((fn) => set(produce(fn)), get, api);

const [useStore] = create(
  devtools(
    immer((set) => ({
      cart: {
        open: false,
        setOpen: () =>
          set((state) => {
            state.cart.open = !state.cart.open;
          }),
      },
    }))
  )
);
```

**Comment by getreworked** [comment]
@clawoflight >However, I use a patch that shows the prefix ('platform') in your example instead of 'instance #'.

What patch did you use for this?

**Comment by dai-shi** [comment]
Can someone help understanding the status of this issue?

**Comment by olavoparno** [comment]
> Can someone help understanding the status of this issue?

Hello, mister dai-shi, as far as I know, this is still an issue. 
Summarizing it all:

- Multiple stores log to Redux Devtools as below's _Oráculo_ and _Instances_ 2 to 5

![image](https://user-images.githubusercontent.com/7513162/90136242-3ed9c900-dd4a-11ea-92ca-9368a9015051.png)

**Comment by dai-shi** [comment]
@olavoparno Thanks for the response.
Unfortunately, I don't have time to investigate into middleware.ts, so we need somebody to help on it.

**Comment by getreworked** [comment]
@dai-shi @olavoparno I've had a bit of time to look into this and found that the devtool `connect` call was missing a name parameter. I've issued a PR for this simple fix solely to fix the instance name issue, but the middleware could be expanded on in the future to accept an options object for even more customization.

Here is my PR: https://github.com/react-spring/zustand/pull/145

And here is the reference to the arguments that redux devtools accepts: https://github.com/zalmoxisus/redux-devtools-extension/blob/master/docs/API/Arguments.md

**Comment by dai-shi** [comment]
Maybe not all issues reported here are solved by #145. Please file a new issue if there's any other issue that should be tracked.
